stant-album/ViewController.swift

```diff
@@ -15,11 +15,28 @@ class ViewController: UIViewController {
     @IBOutlet weak var recordLabel: UILabel!
     @IBOutlet weak var newAlbumButton: UIButton!
     
+    var albumGenerator = AlbumGenerator()
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        getNewAlbum()
     }
 
     @IBAction func getNewAlbumAction(_ sender: Any) {
+        self.getNewAlbum()
+    }
+    
+    func getNewAlbum() {
+        let albumIndex = Int.random(in: 0..<albumGenerator.albumList.count)
+        let newAlbum: Album = albumGenerator.albumList[albumIndex]
+        self.updateFields(newAlbum)
+    }
+    
+    func updateFields(_ album: Album) {
+        self.albumCoverImageView.image = UIImage(named: album.albumCover!)
+        self.albumNameLabel.text = album.albumName!
```

Comment:
1- Identations
2- Always try to avoid force an unwrapping of variables (!). 
In this case, you can use a guard let condition for each setting, or make use of "??" trick, like that: 

self.albumNameLabel.text = album.albumName ?? String()

or

self.albumNameLabel.text = album.albumName ?? ""

In this case, if this albumName is nil, an empty string is setted..

Reply:
Nice! Done.